test.py

```diff
@@ -92,50 +110,58 @@ class MySchema(Schema):
     assert_equal(instance.z, [s.decode('utf8') for s in z])
 
 
-def test_to_dict():
+def test_to_dict(sample_recarray):
     obj = SomeSchema()
     obj.name = 'test'
     obj.x = [1, 2, 3]
+    obj.y = sample_recarray
 
     d = obj.to_dict()
     assert d['name'] == obj.name
     assert_equal(obj.x, d['x'])
+    assert_equal(obj.y, d['y'])
 
 
 @pytest.mark.parametrize('compress', [True, False])
-def test_to_npz(compress, tmpdir):
-    obj = SomeSchema(name='test', x=[1, 2, 3])
+def test_to_npz(compress, tmpdir, sample_recarray):
+    obj = SomeSchema(name='test',
+                     x=[1, 2, 3],
+                     y=sample_recarray)
     path = str(tmpdir.join('test.npz'))
     obj.to_npz(path, compress=compress)
 
     npz = np.load(path)
     assert str(npz['name']) == 'test'
     assert_equal([1, 2, 3], npz['x'])
+    assert_equal(sample_recarray, npz['y'])
 
 
-def test_from_npz(tmpdir):
+def test_from_npz(tmpdir, sample_recarray):
     path = str(tmpdir.join('output.npz'))
-    np.savez(path, x=[1, 2, 3], name='test')
+    np.savez(path, x=[1, 2, 3], name='test', y=sample_recarray)
 
     obj = SomeSchema.from_npz(path)
     assert obj.name == 'test'
     assert_equal([1, 2, 3], obj.x)
+    assert_equal(sample_recarray, obj.y)
 
 
-def test_to_json():
-    obj = SomeSchema(x=list(range(10)), name="whatever")
+def test_to_json(sample_recarray):
+    obj = SomeSchema(x=list(range(10)), name="whatever", y=sample_recarray)
     jobj = obj.to_json()
 
     loaded = json.loads(jobj)
     assert_equal(loaded['x'], obj.x)
+    assert_equal(loaded['y'], obj.y.tolist())
```

Comment:
The objects will not be equal without adding tolist() because numpy does some type coercion as part of calling tolist() on a recarray, which is what is done when this is initially encoding. Is there a better way to do do this?